tests/vector-input.spec.ts

```diff
@@ -0,0 +1,260 @@
+  
+  // x input step default decrements down 1
+  await xInput.focus()
+  await xInput.press('ArrowDown')
+  expect(await xInput.inputValue()).toBe('0')
+
+  // y default step tests
+  const yInput = vectorInputDefault.locator('input').nth(1)
+  expect(await yInput.inputValue()).toBe('')
+
+  await yInput.focus()
+  await yInput.press('ArrowUp')
+  expect(await yInput.inputValue()).toBe('1')
+
+  await yInput.focus()
+  await yInput.press('ArrowDown')
+  expect(await yInput.inputValue()).toBe('0')
+
+  // z default step tests
+  const zInput = vectorInputDefault.locator('input').nth(2)
+  expect(await zInput.inputValue()).toBe('')
+
+  await zInput.focus()
+  await zInput.press('ArrowUp')
+  expect(await zInput.inputValue()).toBe('1')
+
+  await zInput.focus()
+  await zInput.press('ArrowDown')
+  expect(await zInput.inputValue()).toBe('0')
+
+  // input type tests
+  const xTypeValue = await vectorInputDefault.locator('input').nth(0).getAttribute("type")
+  expect(xTypeValue).toBe('text')
+  const yTypeValue = await vectorInputDefault.locator('input').nth(1).getAttribute("type")
+  expect(yTypeValue).toBe('text')
+  const zTypeValue = await vectorInputDefault.locator('input').nth(2).getAttribute("type")
+  expect(zTypeValue).toBe('text')
+
+});
+
+test('Renders three inputs with type text if number type specified', async ({ page }) => {
+  const vectorInputNumType = page.getByTestId('vector-input-type-number')
+
+  await expect(vectorInputNumType).toBeVisible()
+
+  const xTypeValue = await vectorInputNumType.locator('input').nth(0).getAttribute("type")
+  expect(xTypeValue).toBe('text')
+  const yTypeValue = await vectorInputNumType.locator('input').nth(1).getAttribute("type")
+  expect(yTypeValue).toBe('text')
+  const zTypeValue = await vectorInputNumType.locator('input').nth(2).getAttribute("type")
+  expect(zTypeValue).toBe('text')
+
+});
+
+test('Renders three inputs with type number if integer type specified', async ({ page }) => {
+  const vectorInputTypeInteger = page.getByTestId('vector-input-type-integer')
+
+  const xTypeValue  = await vectorInputTypeInteger.locator('input').nth(0).getAttribute("type")
+  expect(xTypeValue ).toBe('number')
+  const yTypeValue = await vectorInputTypeInteger.locator('input').nth(1).getAttribute("type")
+  expect(yTypeValue).toBe('number')
+  const zTypeValue = await vectorInputTypeInteger.locator('input').nth(2).getAttribute("type")
+  expect(zTypeValue).toBe('number')
+
+});
+
+test('When inputs are updated, events fire', async ({ page }) => {
+  const vectorInputOnInput = page.getByTestId('vector-input-on-input')
+
+  await expect(vectorInputOnInput).toBeVisible()
+
+  const xInput = vectorInputOnInput.locator('input').nth(0)
+  const xInputEvent = waitForCustomEventWithParam(page, 'input', 'value')
+  await xInput.focus()
+  await xInput.press('ArrowUp')
+  expect(xInputEvent).toBeTruthy()
+  expect(await getCustomEventParam(page, 'input', 'value')).toBe('1')
```

Comment:
[q] weird that it only dispatches the one value with no annotation if you adjust one dimension? is there no labeling or default for the other dimensions without input?

Reply:
Yeah, that's how it's currently set up - no default for the dimensions without a set input.